[FrameworkBundle] PR #26213 Document redirections with 307/308 HTTP status codes

ZipoKing opened the pull request:
This documents changes introduced with pull request https://github.com/symfony/symfony/pull/26213

ZipoKing committed:
[PR #26213 Document redirections with 307/308 HTTP status codes

Simperfit commented on routing/redirect_in_config.rst:
```diff
@@ -155,3 +155,76 @@ action:
+Redirecting POST/PUT calls
+--------------------------
+
+As default behaviour of both methods mentioned above results in sending
```
`As a`

Simperfit commented on routing/redirect_in_config.rst:
```diff
@@ -155,3 +155,76 @@ action:
+
+        return $collection;
+
+Switching ``keepRequestMethod`` switch to ``true`` will result in sending
```
When ``keepRequestMethod`` is set to true with either ``permanent`` set to false which will lead to a ``307`` response or ``308`` with ``permanent`` being true. Theses codes will give information in the HTTP request that the method should be repeated without altering the body.

Simperfit commented on routing/redirect_in_config.rst:
```diff
@@ -155,3 +155,76 @@ action:
+        use Symfony\Component\Routing\Route;
+
+        $collection = new RouteCollection();
+        // ...
```
you can remove this

Simperfit commented on routing/redirect_in_config.rst:
```diff
@@ -155,3 +155,76 @@ action:
+            xsi:schemaLocation="http://symfony.com/schema/routing
+                http://symfony.com/schema/routing/routing-1.0.xsd">
+
+            <!-- ... -->
```
you can remove this

Simperfit commented on routing/redirect_in_config.rst:
```diff
@@ -155,3 +155,76 @@ action:
+
+        # config/routes.yaml
+
+        # ...
```
you can remove this

Simperfit commented on routing/redirect_in_config.rst:
```diff
@@ -155,3 +155,76 @@ action:
+
+As default behaviour of both methods mentioned above results in sending
+response with ``301`` or ``302`` HTTP status codes, the following call will
+be made with use of HTTP request method. But it some scenarios it is
```
`use of HTTP request method`
WDYM ? 

`But it some scenarios`
But in some scenarios it's either expected or required that the following call [...]

ZipoKing committed:
Changes after @Simperfit  review

ZipoKing commented:
Thank you @Simperfit - all changes applied

javiereguiluz committed:
Reword and added an example

I did some rewords based on https://symfony.com/blog/new-in-symfony-4-1-307-and-308-redirections and added a full config example.

javiereguiluz commented:
@ZipoKing thanks for contributing the docs for this nice feature that you also contributed ... and congrats on your first Symfony Docs contribution!
